modules/miner/blockmanager.go

```diff
@@ -158,6 +162,14 @@ func (m *Miner) managedSubmitBlock(b types.Block) error {
 
 // SubmitHeader accepts a block header.
 func (m *Miner) SubmitHeader(bh types.BlockHeader) error {
+	if err := m.tg.Add(); err != nil {
```

Comment:
`IsStopped` check should happen before you add to the `tg`, and since this function returns an error you should return `errMinerClosed` if the miner is closed.